shared/src/commonMain/kotlin/io/customer/shared/sdk/config/CustomerIOConfig.kt

```diff
@@ -0,0 +1,29 @@
+package io.customer.shared.sdk.config
+
+import io.customer.shared.sdk.meta.Workspace
+import io.customer.shared.util.LogLevel
+
+/**
+ * Config class to hold SDK configurations in a single place.
+ *
+ * @property logLevel the level of logs to print.
+ * @property workspace configurations required to setup the workspace.
+ * @property network configurations to hold network settings.
+ * @property backgroundQueue configurations to hold background queue settings.
+ * @property moduleConfig map to hold module configurations, having a map it makes it easier to
+ * attach them later.
+ */
+data class CustomerIOConfig(
```

Comment:
Broke down CustomerIOConfig into multiple configuration classes considering the following points in mind
- For easier access and limiting the usage to relevant classes
- Assuming every layer and/or module have their independent configuration
- Implemented builder pattern so it is easier to provide only required values from Java classes as well

Open Questions?
- How much the impact will be on SDK classes and tests?
- Is this too much splitting? If yes, then why (other than the refactoring impact)?
- How will the the builder pattern behave for iOS SDK?

Reply:
I think we have two major questions here

> what is the reason for adding CustomerIOConfig to KMM in this PR

If we take a look at current [`CustomerIOConfig`](https://github.com/customerio/customerio-android/blob/develop/sdk/src/main/java/io/customer/sdk/CustomerIOConfig.kt), around 8 out of 12 properties are mainly for running queue. And at some point, we need to access them in KMM somehow. Out of all the solutions I tried to solve this, moving `CustomerIOConfig` to KMM looked the most balanced solution for this. Also, this means when we move to KMM BQ, the usage of `CustomerIOConfig` in our SDKs will be very limited as the classes using most of these properties might already be removed with BQ V1 code removal.

> Should we breakdown the CustomerIOConfig class into separate classes?

I already explained above that it's just a proposal and added the thinking behind breaking it down. Every time I have to add some configuration to this, I feel like we are overloading it slightly. Also, adding module wide configurations were even more difficult. We previously added map to support dynamic module wide configurations in Android SDK, and I would just like to improve it more with these changes.

But if you think we don't need this now, that's completely fine. Please feel free to share your thoughts and why you think we should avoid that. I'm completely okay for holding changes and the discussion for later phases.

To answer your other questions,

I think it is only `public` because we needed to share it between modules. I don't see a use case where customers might need to read values from it. Also, it is not forcing to change any customer facing apis as we take these params in builder and build this internally in Android. For iOS, we can keep a mapping to bridge these changes with some deprecated methods. But its completely okay to skip for now, we can keep this for later.